src/server.js

```diff
@@ -202,12 +204,107 @@ app.get(
   }
 );
 
+function averageGeolocation(coords) {
+  if (coords.length === 1) {
+    return coords[0];
+  }
+  let x = 0.0;
+  let y = 0.0;
+  let z = 0.0;
+
+  for (let coord of coords) {
+    let latitude = (coord.latitude * Math.PI) / 180;
+    let longitude = (coord.longitude * Math.PI) / 180;
+
+    x += Math.cos(latitude) * Math.cos(longitude);
+    y += Math.cos(latitude) * Math.sin(longitude);
+    z += Math.sin(latitude);
+  }
+
+  let total = coords.length;
+
+  x = x / total;
+  y = y / total;
+  z = z / total;
+
+  let centralLongitude = Math.atan2(y, x);
+  let centralSquareRoot = Math.sqrt(x * x + y * y);
+  let centralLatitude = Math.atan2(z, centralSquareRoot);
+
+  return {
+    latitude: (centralLatitude * 180) / Math.PI,
+    longitude: (centralLongitude * 180) / Math.PI,
+  };
+}
+
+let sessionName = "Get together";
+
+// Mock data; expect ~ 37.790831, -122.407169
+const sf = [
+  {
+    latitude: 37.797749,
+    longitude: -122.412147,
+  },
+  {
+    latitude: 37.789068,
+    longitude: -122.390604,
+  },
+  {
+    latitude: 37.785269,
+    longitude: -122.421975,
+  },
+];
+
+// Mock data; expect ~ 8.670552, -173.207864
+const globe = [
+  {
+    // Japan
+    latitude: 37.928969,
+    longitude: 138.979637,
+  },
+  {
+    // Nevada
+    latitude: 39.029788,
+    longitude: -119.594585,
+  },
+  {
+    // New Zealand
+    latitude: -39.298237,
+    longitude: 175.717917,
+  },
+];
+
+app.get(`${PREFIX_API}/:${URL_PARAM_EVENT_ID}/restaurants`, function () {
+  getEvent,
+    async (request, response) => {
+      const { event } = request;
+      const users = event.users || {};
+      response.json({
+        status: 200,
+        data: Object.values(users),
+      });
+    };
+
+  const query = datastore.createQuery("Company");
+  let location = averageGeolocation(sf).toString();
+  let lat = location.latitude.toString();
+  let long = location.longitude.toString();
+  let radius = "50000";
+  let type = "restaurant";
+  let minprice = "0";
+  let maxprice = "4";
+  fetch(
+    `https://maps.googleapis.com/maps/api/place/nearbysearch/json?location=${lat},${long}&radius=${radius}&type=${type}&minprice=${minprice}&maxprice=${maxprice}&key=APIkey`
```

Comment:
`key=APIKey` will send the api key as `APIKey`. You probably want to use string interpolation, like so: `key=${APIKey}` (assuming `APIKey` is a variable, etc. etc.)